main.tf

```diff
@@ -1 +1,381 @@
+  bucket = aws_s3_bucket.eks_audit_log_bucket.id
+  versioning_configuration {
+    status     = local.bucket_enable_versioning
+    mfa_delete = local.mfa_delete
+  }
+}
+
+resource "aws_s3_bucket_notification" "bucket_notification" {
+  bucket = aws_s3_bucket.eks_audit_log_bucket.id
+
+  topic {
+    topic_arn     = aws_sns_topic.eks_sns_topic.arn
+    events        = ["s3:ObjectCreated:*"]
+    filter_prefix = "eks_audit_logs/${data.aws_caller_identity.current.account_id}"
+  }
+  depends_on = [aws_sns_topic.eks_sns_topic, aws_sns_topic_policy.eks_sns_topic_policy]
+}
+
+resource "aws_iam_role" "firehose_iam_role" {
+  name               = local.firehose_iam_role_name
+  assume_role_policy = data.aws_iam_policy_document.firehose_iam_assume_role_policy.json
+}
+
+data "aws_iam_policy_document" "firehose_iam_assume_role_policy" {
+  version = "2012-10-17"
+
+  statement {
+    sid     = "LaceworkEKSFirehoseIAMRole"
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["firehose.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_policy" "firehose_iam_policy" {
+  name        = local.firehose_policy_name
+  description = "A firehose IAM policy"
+  policy      = data.aws_iam_policy_document.firehose_iam_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "firehose_iam_role_policy" {
+  role       = local.firehose_iam_role_name
+  policy_arn = aws_iam_policy.firehose_iam_policy.arn
+  depends_on = [aws_iam_policy.firehose_iam_policy]
+}
+
+data "aws_iam_policy_document" "firehose_iam_role_policy" {
+  version = "2012-10-17"
+
+  statement {
+    sid = "LaceworkEKSFirehoseIAMRole"
+    actions = [
+      "s3:AbortMultipartUpload",
+      "s3:GetBucketLocation",
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:ListBucketMultipartUploads",
+      "s3:PutObject",
+    ]
+    effect    = "Allow"
+    resources = ["${aws_s3_bucket.eks_audit_log_bucket.arn}/*"]
+  }
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
+  name        = local.firehose_delivery_stream_name
+  destination = "extended_s3"
+
+  extended_s3_configuration {
+    role_arn            = aws_iam_role.firehose_iam_role.arn
+    bucket_arn          = aws_s3_bucket.eks_audit_log_bucket.arn
+    prefix              = "eks_audit_logs/${data.aws_caller_identity.current.account_id}/"
+    buffer_interval     = 300
+    buffer_size         = 100
+    error_output_prefix = "audit_logs/${data.aws_caller_identity.current.account_id}/error/"
+    compression_format  = "UNCOMPRESSED"
+    cloudwatch_logging_options {
+      enabled = false
+    }
+  }
+  server_side_encryption {
+    enabled = false
+  }
+  depends_on = [aws_s3_bucket.eks_audit_log_bucket, aws_iam_role.firehose_iam_role]
+}
+
+module "lacework_eks_audit_iam_role" {
+  source                  = "lacework/iam-role/aws"
+  version                 = "~> 0.1"
+  create                  = true
```

Comment:
Do we always create an IAM role?

Reply:
ok

Reply:
I'm ok with this for the time being, but we probably want to allow re-use of our existing role.